From 8c3e6373a7742f50686fd96013fecd5085c46716 Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 21 Mar 2022 15:11:12 +0200
Subject: [PATCH] [SW-43475] greco: arc: move preboot load address to start of sram

This patch changes preboot start address to start of sram, instead of
16KB offset, which was previously left by mistake for ppboot.
Ppboot loads in fw sram, so sram can directly start from 0 offset.

Also it is seem that while starting fro 16K offset, max 48K is getting
copied into sram from spi flash using dma, since preboot size can be
greater then 48K, it causes mismatch in copied data.
When starting from 0th offset, more then 48K is also getting copied
fine.

Change-Id: I2740b433f5eef57de65506383a19c26cee88ea7d
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/board/habanalabs/greco/agent-fw_loader.its b/board/habanalabs/greco/agent-fw_loader.its
index 4daf926..b7a021c 100644
--- a/board/habanalabs/greco/agent-fw_loader.its
+++ b/board/habanalabs/greco/agent-fw_loader.its
@@ -105,7 +105,7 @@
 			arch = "arc";
 			compression = "none";
 			offset = <0x00020000>, <0x00040000>;
-			loadaddr = <0x7F 0xF0004000>;
+			loadaddr = <0x7F 0xF0000000>;
 			hash {
 				algo = "sha256";
 			};
